cmd/soroban-cli/src/commands/network/container/logs.rs

```diff
@@ -15,17 +17,19 @@ pub enum Error {
 
 #[derive(Debug, clap::Parser, Clone)]
 pub struct Cmd {
-    /// Network to tail
-    pub network: Network,
+    #[command(flatten)]
+    pub container_args: Args,
 
-    #[arg(short = 'd', long, help = DOCKER_HOST_HELP, env = "DOCKER_HOST")]
-    pub docker_host: Option<String>,
+    /// Network container to tail (used in container name generation)
+    #[arg(required_unless_present = "container_name")]
+    pub network: Option<Network>,
 }
 
 impl Cmd {
     pub async fn run(&self) -> Result<(), Error> {
-        let container_name = format!("stellar-{}", self.network);
-        let docker = connect_to_docker(&self.docker_host).await?;
+        let container_name =
+            get_container_name(self.container_args.container_name.clone(), self.network);
```

Comment:
```suggestion
            get_container_name(self.container_args.container_name.as_deref(), self.network);
```
This will let you do `Option<String>` --> `Option<&str>`, and remove the need to clone (once the parameter is updated).